### DISCUSSION TITLE
[Proposal / Implementation Help] Exponential Backoff in API throttling

### DISCUSSION THREAD

**Discussion by Gicminos** [laravel/framework#33028] [General]
Dear all and thank you to read my idea. 

I was wondering if implementing an exponential backoff on the laravel throttle middleware could be something interesting to be implemented into the core. If not I would love to hear some suggestions on how to ovverride it or implement it differently.

I have searched for long online and I haven't found anything regarding it. At this point I am also wondering if my idea could be a bad implementation.

Let me provide an example:

- A user is initally blocked for 1 minute
- If The user reaches the limit again in the 1 minute after, he is then blocked for 2 minutes
- If all this happens again, the user is blocked for 4 minutes
- If the user behave well in the next 4 minutes, the block gets reduced to 2 and so on

**Comment by DarkGhostHunter** [comment]
There is a new Rate Limiter for Laravel 8.0 that should allow you to do that.

**Reply by Gicminos** [reply]
Thank you for your answer! That is very cool to hear and increases my interest in Laravel 8.0 that already looks promising. Should we leave this discussion open until Laravel 8.0 is released and more documentation will be provided to do so?
